src/Utilities/WellKnownTypeProvider.cs

```diff
@@ -67,6 +73,68 @@ private WellKnownTypeProvider(Compilation compilation)
         /// </summary>
         public INamedTypeSymbol GenericIEquatable { get; }
 
+        /// <summary>
+        /// <see cref="INamedTypeSymbol"/> for <see cref="System.Runtime.Serialization.Formatters.Binary.BinaryFormatter"/>
+        /// </summary>
+        public INamedTypeSymbol BinaryFormatter { get; }
+
+        /// <summary>
+        /// <see cref="INamedTypeSymbol"/> for <see cref="System.Web.UI.LosFormatter"/>
+        /// </summary>
+        public INamedTypeSymbol LosFormatter { get; }
+
+        /// <summary>
+        /// Mapping of <see cref="ISymbol"/> to full name (e.g. "System.Exception").
+        /// </summary>
+        private Dictionary<ISymbol, string> TypeToFullName { get; }
+
+        /// <summary>
+        /// Mapping of full name to <see cref="INamedTypeSymbol"/>.
+        /// </summary>
+        private Dictionary<string, INamedTypeSymbol> FullNameToType { get; }
+
+        /// <summary>
+        /// Attempts to get the full type name (namespace + type) of the specifed symbol.
+        /// </summary>
```

Comment:
Is there a good way to go from an INamedTypeSymbol to a qualified name?  Just wondering if I'm missing something and this is silly...

Reply:
Something like [https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticAnalyzerCorrectnessAnalyzer.cs#L9](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticAnalyzerCorrectnessAnalyzer.cs#L9) ?

---
In reply to: [223148829](https://github.com/dotnet/roslyn-analyzers/pull/1838#discussion_r223148829) [](ancestors = 223148829)

Reply:
Ah, sorry misread your question. What you need is probably [http://source.roslyn.io/#Microsoft.CodeAnalysis/SymbolDisplay/SymbolDisplayFormat.cs,197](http://source.roslyn.io/#Microsoft.CodeAnalysis/SymbolDisplay/SymbolDisplayFormat.cs,197), which is internal

---
In reply to: [223151018](https://github.com/dotnet/roslyn-analyzers/pull/1838#discussion_r223151018) [](ancestors = 223151018,223148829)

Reply:
I'm removing that method for now, and reviving if I need it for tainted data analysis.